backend/src/bundles/templates/template.repository.ts

```diff
@@ -20,8 +21,33 @@ class TemplateRepository implements ITemplateRepository {
         return await this.templateModel.query().findById(id);
     }
 
-    public async findAll(): Promise<TemplateGetAllResponseDto> {
-        const response = await this.templateModel.query();
+    public async findAll(
+        options?: FindAllOptions,
+    ): Promise<TemplateGetAllResponseDto> {
+        let query = this.templateModel.query();
+
+        if (options && options.sortBy) {
+            const [field, order] = options.sortBy.split(':');
+            if (
+                field &&
+                order &&
+                ['asc', 'desc'].includes(order.toLowerCase())
+            ) {
+                query = query.orderBy(
+                    field,
+                    order.toLowerCase() as OrderByDirection,
+                );
+            } else {
+                throw new Error('Invalid sorting criteria. Use field:order');
```

Comment:
throw HTTPError with 401 Bad credentials

Reply:
401 for wrong query format?)
400 may be better??

Reply:
ye